### DISCUSSION TITLE
I want to use Oauth2 to signup new user and automatically login but I set API rules of create user to adminOnly. It end up creating only a new user but not login.

### DISCUSSION THREAD

**Discussion by idea2547** [pocketbase/pocketbase#4061] [Q&A]
`await adminClient.collection('users').authWithOAuth2(
            provider.name, code, expectedVerifier, redirectURL 
 );`
AdminClient pocketbase instance Oauth2 google login only update new user on DB but not save authData of user

 `const authData = await locals.pb.collection('users').authWithOAuth2(
            provider.name, code, expectedVerifier, redirectURL 
 );`
this locals.pb login successfully save user authData but cannot create new user due to adminOnly CREATE Api rules

**Comment by ganigeorgiev** [comment]
I'm failing to understand the description of the problem, what request is being send and what is the desired outcome.

If you want to allow users to create accounts via OAuth2 but to disallow requests to the create API, you can try to enable the Create API rule but also define an API hook that will disallow requests from non-admins. For example:

- [Go](https://pocketbase.io/docs/go-overview/):
    ```go
    app.OnRecordBeforeCreateRequest("users").Add(func(e *core.RecordCreateEvent) error {
        admin, _ := e.HttpContext.Get(apis.ContextAdminKey).(*models.Admin)
        if admin != nil {
            return nil // allow for admins
        }

        return apis.NewForbiddenError("User create requests are not allowed", nil)
    })
    ```

- [JS `pb_hooks`](https://pocketbase.io/docs/js-overview/):
    ```js
    // pb_hooks/main.pb.js

    onRecordBeforeCreateRequest((e) => {
        const admin = e.httpContext.get("admin")
        if (!admin) {
            throw new ForbiddenError("User create requests are not allowed");
        }
    }, "users")
    ```
